### DISCUSSION TITLE
Feature Request: New API Additions (Round 3)

### DISCUSSION THREAD

**Discussion by MattTheCuber** [HypixelDev/PublicAPI#642] [Ideas]
Hello Hypixel Developers,

This is a follow-up of https://github.com/HypixelDev/PublicAPI/discussions/542 and https://github.com/HypixelDev/PublicAPI/discussions/572.

Thank you Hypixel developers for all that you do to support us developers. Your API allows us to create valuable tools for players. As part of the SkyHelper development team, I've been utilizing your API for many years now. In this discussion, we have compiled a list of requested features, grouped by how critical they are to players and developers to build useful tools. While we understand some may be too complex to integrate, it would be amazing if you could spend some time going through each item. I would suppose that **many** items are simple one-line changes to your API to expose the respective key. Others might be more challenging, but any information is better than none. If some of these items are definitely impossible or not going to be exposed, please tell us. Also, do not feel like you need to answer this discussion all in one comment. I can continue to update this post as you expose features in the API. Your efforts are greatly appreciated by the SkyBlock community. Thank you for considering our requests.

## Most Critical Features
- ~~**Taming Level Cap**~~ Added ✅
The most important (and probably easiest to add) feature we desperately need is a way to get the taming level cap of a profile. Having been introduced a couple months ago, it is not possible since then to calculate a reliable taming level or skill average.

- ~~**Garden Data**~~ Added (*mostly*) ✅
It has been over a year since the garden update was introduced to SkyBlock and it would be nice to be able to access data such as garden level experience, crop upgrade levels, garden milestones, plot configuration and visitor data on a new skyblock/garden endpoint like it was added with the museum. Here is a discussion with a list of missing Garden data: https://github.com/HypixelDev/PublicAPI/discussions/618.

- ~~**Personal Bank**~~ Added ✅
This feature was added in the first feature request post. However, soon after it was removed because of https://github.com/HypixelDev/PublicAPI/issues/544. It would be really cool if this could be added back with proper API disabling.

- **Jacob's Contests**
For years we rely on mods collecting Jacob Contests from when player's open their calendar. After more than 3 years it would be nice to have them in a dedicated endpoint.

## Highly Requested Missing Features
- ~~**Booster Cookie Active Flag**~~ Added ✅
Time and time again there is need to know if someone has a booster cookie active. We have requested this before, but if it would be possible just to have a true/false flag that would be fantastic. This could even be API togglable in the SkyBlock menu.

- **Kat**
It would be very cool to see what pet and how much time is left for a player profile's current Kat upgrade (if in use).

- **Fann Pet Trainer**
It would be nice being able to see a player's current trained pets.

- **Hoot Upgrades**
It would great to have a simple key that tells you your hoot upgrade level.

- **Missing Consumable Items**
There are currently ~~two~~ one consumable items that give the player SkyBlock experience or other permanent stats that are missing in the API in order to more accurately implement SkyBlock level progress and related QoL features (~~Wriggling Larva and~~ (added ✅) Refined Dark Cacao Truffle).

- **North Stars**
Just a simple key that shows the number of north stars on the profile.

- **Pandora's Box**
Would be useful to see a player's progress and completions on the Pandora's Box challenges.

- ~~**Crimson Isle Quests**~~ Added ✅
Those used to be in the API but were later removed for an unknown reason. If there is nothing against those we would like to see them back in the API.

- ~~**Sack of Sacks**~~ Added ✅ 
We have sack contents, but it would be cool to know what sacks they own and how many pocket in a sack are applied.

- **City Project Buffs**
Very simple just a list of city project and what bonus the player received in that project. Mostly useful for finding discounts on shops and slayer items.

- **Perkpocalypse Mayor Order**
It would be amazing if the /skyblock/mayor endpoint showed the order of Perkpocalyse mayors when Mayor Jerry is elected.

- **Carnival Mask Bag**
We do not have a way to access the masks contained by a carnival mask bag in the API. I assume it would look something like New Year's Cake Bag data.

- ~~**Buyer Profile in Auctions Ended Endpoint**~~ Added ✅ 
See https://github.com/HypixelDev/PublicAPI/discussions/545.

- **RNG Meter**
It would be cool if the API had a section for the profile's RNG meter selection and information.

- **Bingo Rank**
It would be really handy to have a key in the bingo profile endpoint to see a player's bingo rank.

- **Pet Editions**
We would love it if you could include an edition number within the Eerie pet data received from the API response. This would help in determining which edition the player has acquired.

- **More Kuudra Data**
If possible, having similar data to what we have with dungeons would be useful such as fastest run, damage dealt and more.

- **Individual Fairy Souls Collected**
Having a list of the individual fairy souls collected would be amazing as then people could determine exactly which fairy souls they are missing and not have to check dozens until they find it.

- **Defuse Kit Bombs Defused**
Just a simple key that shows the number of bombs defused by defuse kits on the profile.

## Other/New Endpoints
- **Bazaar Orders Endpoint**
Similar to the [`/auctions` endpoint](https://api.hypixel.net/#tag/SkyBlock/paths/~1v2~1skyblock~1auction/get), it would amazing if you could view a player's bazaar orders.

- **New `/skyblock/resource` Endpoints**
Almost all developers need to manually collect and store "constant" information from the game such as pets and enchantments. It would be really nice if there were resource endpoints for each of these like `/items` (/pets, /bestiary, /enchantments, /skins, /petskins, /recipes, /trades, /mobs, /museum, /reforges, etc.). See https://github.com/NotEnoughUpdates/NotEnoughUpdates-REPO for more ideas.

- **City Projects Endpoint**
It would be cool if there were an endpoints for viewing the active/ended city project and related information.

- **SkyBlock Calendar**
It is possible to calculate SkyBlock events but when new events are added or one-time events are happening it would be nice having an endpoint where those can be retrieved.

- **Remaining Fire Sale Items**
It would be nice to know how many items remain in fire sales on the fire sales endpoints.

- **Placed Minions**
Number and types of placed minions. Possibly also the inventory of those minions such as upgrades and fuel.

- **Island Beacon**
Placed Island Beacon and current configuration.

Note for other developers: Feel free to suggest additions to this list in the comments below. Also, if we have already learned that something isn't going to/can't be added to the API, please let me know so I can remove it from the list.

**Comment by MattTheCuber** [comment]
Thank you very much for the additions today. I crossed off the personal bank, booster cookie, and buyer profile requests as those seem to be implemented. Hopefully you will be able to implement more items from this list! Thank you again for supporting the community in this way, it helps a lot!

**Comment by csjh** [comment]
The ideal end-all-be-all bazaar endpoint for me (from the perspective of a relatively large bazaar website) would be a log of all transactions (instabuy, instasell, order placed, order removed) that happened in the last 60 seconds. This would allow API consumers to effectively recreate a local copy of the entire bazaar if they have the DB capacity, plus as many details as they want in terms of what happened when.
